Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Snapping/snap.h

```diff
@@ -697,6 +727,32 @@ std::size_t split_edges(EdgesToSplitContainer& edges_to_split,
       std::cout << "Actually split? " << do_split << std::endl;
 #endif
 
+      // check the new faces after split will not be degenerated
+      Point p0 = new_position;
+      Point_ref p1 = get(vpm_T, source(h_to_split, tm_T));
+      Point_ref p2 = get(vpm_T, target(next(opposite(h_to_split, tm_T), tm_T), tm_T));
+      Point_ref p3 = get(vpm_T, target(h_to_split, tm_T));
+
+      /* Chooses the diagonal that will split the quad in two triangles that maximize
+       * the scalar product of of the un-normalized normals of the two triangles.
+       * The lengths of the un-normalized normals (computed using cross-products of two vectors)
+       *  are proportional to the area of the triangles.
+       * Maximize the scalar product of the two normals will avoid skinny triangles,
+       * and will also taken into account the cosine of the angle between the two normals.
+       * In particular, if the two triangles are oriented in different directions,
+       * the scalar product will be negative.
+       */
+      auto p1p3 = CGAL::cross_product(p2-p1,p3-p2) * CGAL::cross_product(p0-p3,p1-p0);
+      auto p0p2 = CGAL::cross_product(p1-p0,p1-p2) * CGAL::cross_product(p3-p2,p3-p0);
+      bool first_split_face = (p0p2>p1p3);
+
+      bool is_deg = first_split_face
+                  ? collinear(p0,p1,p2) && collinear(p0,p3,p2)
+                  : collinear(p0,p1,p3) && collinear(p1,p2,p3);
+
+      if (is_deg)
```

Comment:
Like at other places, I'm more in favor of doing the snap and treating the degenerate faces (it's a split of a border halfedge, it should be rather straightforward?)